uefi/src/fs/path/mod.rs

```diff
@@ -0,0 +1,53 @@
+//! This module offers the [`Path`] and [`PathBuf`] abstractions.
+//!
+//! # Interoperability with Rust strings
+//!
+//! For the interoperability with Rust strings, i.e., `String` and `str` from
+//! the standard library, the API is intended to transform these types first to
+//! `CString16` respectively `CStr16`. They do not directly translate to
+//! [`Path`] and [`PathBuf`].
+//!
+//! # Path Structure
+//!
+//! Paths use the [`SEPARATOR`] character as separator. Paths are absolute and
+//! do not contain `.` or `..` components. However, this can be implemented in
+//! the future.
+
+mod path;
+mod pathbuf;
+mod validation;
+
+pub use path::Path;
+pub use pathbuf::PathBuf;
+
+use uefi::{CStr16, Char16};
+pub(super) use validation::validate_path;
+pub use validation::PathError;
+
+/// The default separator for paths.
+pub const SEPARATOR: char = '\\';
+/// [`SEPARATOR`] but as useful UEFI type.
+pub const SEPARATOR_C16: Char16 = unsafe { Char16::from_u16_unchecked('\\' as u16) };
+
+/// Stringified version of [`SEPARATOR`].
+pub const SEPARATOR_STR: &str = "\\";
+/// [`SEPARATOR_STR`] but as useful UEFI type.
+pub const SEPARATOR_CSTR16: &CStr16 = uefi_macros::cstr16!("\\");
```

Comment:
I'm not sure if the `SEPARATOR_STR` and `SEPARATOR_CSTR16` constants are needed. (Commented elsewhere about converting a use of `SEPARATOR_CSTR16` -> `SEPARATOR_C16`). I know std has https://doc.rust-lang.org/std/path/constant.MAIN_SEPARATOR_STR.html, but I can't think of why one would use that instead of the char one.

Reply:
I also do not have a strong opinion here. Probably we can drop them, yes